src/PresentationalComponents/Notifications/utils.js

```diff
@@ -1,66 +1,104 @@
-import { BULK_SELECT_BUTTON } from '../../Utilities/constants';
-import config from '../../config/config.json';
+import omit from 'lodash/omit';
+import {
+  BULK_SELECT_BUTTON,
+  INPUT_GROUP,
+  TAB_GROUP,
+} from '../../SmartComponents/FormComponents/componentTypes';
 
 export const prepareFields = (notifPref, emailPref, emailConfig) =>
-  Object.entries(notifPref).reduce(
-    (acc, [key, value]) => [
+  Object.entries(notifPref).reduce((acc, [bundleKey, bundleData]) => {
+    return [
       ...acc,
       {
-        title: config['notification-preference'][key].title,
-        name: key,
-        fields: value.sections.reduce((acc, section) => {
-          let unsubscribeAllInactive = true;
-          const fields = [
-            ...section.fields[0].fields.map((field) => {
-              unsubscribeAllInactive =
-                unsubscribeAllInactive && field.initialValue;
-              return {
-                ...field,
-                group: key,
-                section: section.name,
-                category: 'notification-preference',
-              };
-            }),
-            ...Object.entries(emailPref).reduce(
-              (acc, emailSection) => [
-                ...acc,
-                ...(emailSection[0] === section.name &&
-                emailConfig[emailSection[0]]?.bundle === key
-                  ? emailSection?.[1]?.schema?.[0]?.fields?.map((field) => {
-                      unsubscribeAllInactive =
-                        unsubscribeAllInactive && field.initialValue;
+        title: bundleData?.label,
+        name: bundleKey,
+        fields: Object.entries(bundleData.applications).reduce(
+          (acc, [appKey, appData]) => {
+            let selectAllActive = true;
```

Comment:
This is fairly complicated, we are probably out of luck with something less complex, right?

Reply:
this is the best I've been able to come up with :laughing: 
If we don't want the backend to add more UI details to the schema, we need to loop through it and wrap it with the remaining components.